In [0]:
import pandas as pd
import pandas
import requests
import json
import urllib 
from urllib.request import Request, urlopen

In [0]:
from google.colab import drive
drive.mount('/content/drive')
df = pd.read_csv(open('drive/My Drive/Colab Notebooks/msas.csv', 'rb'))

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [0]:
df.head()

CBSA                         NAME  POPESTIMATE2017
0  10180                  Abilene, TX           170219
1  10420                    Akron, OH           703505
2  10500                   Albany, GA           151434
3  10540                   Albany, OR           125047
4  10580  Albany-Schenectady-Troy, NY           886188

In [0]:
geoid = '31000US'+ df['CBSA'].astype(str)
Half = geoid[:len(geoid)//2]
Res = geoid[len(geoid)//2:]

Half = ','.join(Half)
Res = ','.join(Res)

Question 1: What city has the higest count of the audience you chose for parameter

In [0]:
tid = ['B19001']
tstring =','.join(tid)

In [0]:
reqtur='http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tstring,Half)
loadjson = requests.get(reqtur)
parsjson1 = json.loads(loadjson.text)

reqtur='http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tstring,Res)
loadjson = requests.get(reqtur)
parsjson2 = json.loads(loadjson.text)


In [0]:
HH=pd.DataFrame(columns=['country','population'], index=list(range(1,len(geoid))))

In [0]:
for i in list(range(1,len(geoid)//2)):
  HH['country'][i]=[geoid[i]]
  HH['population'][i]=parsjson1['data'][geoid[i]]['B19001']['estimate']['B19001011']+parsjson1['data'][geoid[i]]['B19001']['estimate']['B19001012']

In [0]:
for i in list(range(192,len(geoid))):
  HH['country'][i]=[geoid[i]]
  HH['population'][i]=parsjson2['data'][geoid[i]]['B19001']['estimate']['B19001011']+parsjson2['data'][geoid[i]]['B19001']['estimate']['B19001012']

In [0]:
print(HH.loc[HH['population']== max(HH['population'])])

            country   population
249  [31000US35620]  1.01121e+06


In [0]:
print(df.loc[df['CBSA']== 35620])

      CBSA                                   NAME  POPESTIMATE2017
249  35620  New York-Newark-Jersey City, NY-NJ-PA         20320876


New York-Newark-Jersey City has the higest count of the audience you chose for parameter. The range between 50K to 74.9 K with 1.01121e+06

Question 2: What city has the higest count of audience you chose for parameter

In [0]:
tid2=['B01001']
tstring2=','.join(tid2)

In [0]:
reqtur='http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tstring2,Half)
loadjson = requests.get(reqtur)
parsjson21 = json.loads(loadjson.text)

reqtur='http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tstring2,Res)
loadjson = requests.get(reqtur)
parsjson22 = json.loads(loadjson.text)

In [0]:
age = ['B01001007','B01001008','B01001009','B01001010','B01001011','B01001012']

In [0]:
df1 = pd.DataFrame(columns=['location','population'],index=list(range(1,len(geoid))))

In [0]:
for i in list(range(1,len(geoid)//2)):
  df1['location'][i]=[geoid[i]]
  df1['population'][i]=parsjson21['data'][geoid[i]]['B01001']['estimate'][age[0]]+parsjson21['data'][geoid[i]]['B01001']['estimate'][age[1]]+parsjson21['data'][geoid[i]]['B01001']['estimate'][age[2]]+parsjson21['data'][geoid[i]]['B01001']['estimate'][age[3]]+parsjson21['data'][geoid[i]]['B01001']['estimate'][age[4]]+parsjson21['data'][geoid[i]]['B01001']['estimate'][age[5]]

In [0]:
for i in list(range(192,len(geoid))):
  df1['location'][i]=[geoid[i]]
  df1['population'][i]=parsjson22['data'][geoid[i]]['B01001']['estimate'][age[0]]+parsjson22['data'][geoid[i]]['B01001']['estimate'][age[1]]+parsjson22['data'][geoid[i]]['B01001']['estimate'][age[2]]+parsjson22['data'][geoid[i]]['B01001']['estimate'][age[3]]+parsjson22['data'][geoid[i]]['B01001']['estimate'][age[4]]+parsjson22['data'][geoid[i]]['B01001']['estimate'][age[5]]

In [0]:
print(df1.loc[df1['population']== max(df1['population'])])

           location   population
249  [31000US35620]  2.33054e+06


In [0]:
print(df.loc[df['CBSA']== 35620])

      CBSA                                   NAME  POPESTIMATE2017
249  35620  New York-Newark-Jersey City, NY-NJ-PA         20320876


New York-Newark-Jersey City has the higest count of age range from 18 to 34 with 2.33054e+06

Qustion 3: What city has the highest count of the audience you choose for parameter.

In [0]:
tid3=['B03003']
tstring3 = ','.join(tid3)

In [0]:
reqtur='http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tstring3,Half)
loadjson = requests.get(reqtur)
parsjson31 = json.loads(loadjson.text)

reqtur='http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tstring3,Res)
loadjson = requests.get(reqtur)
parsjson32 = json.loads(loadjson.text)

In [0]:
df2 = pd.DataFrame(columns=['location','population'],index=list(range(1,len(geoid))))

In [0]:
for i in list(range(1,len(geoid)//2)):
  df2['location'][i]=[geoid[i]]
  df2['population'][i]=parsjson31['data'][geoid[i]]['B03003']['estimate']['B03003003']

In [0]:
for i in list(range(192,len(geoid))):
  df2['location'][i]=[geoid[i]]
  df2['population'][i]=parsjson32['data'][geoid[i]]['B03003']['estimate']['B03003003']

In [0]:
print(df2.loc[df2['population']== max(df2['population'])])

           location   population
210  [31000US31080]  6.00354e+06


In [0]:
print(df.loc[df['CBSA']==31080])

      CBSA                                NAME  POPESTIMATE2017
210  31080  Los Angeles-Long Beach-Anaheim, CA         13353907


Los Angeles-Long Beach-Anaheim has the highest count of hispanic with 6.00354e+06

Question 3: What city has the highest mean percentage across all three categories?

In [0]:
percentage_income = []
percentage_age = []
percentage_hispanic = []

In [0]:
for i, row in HH.iterrows():
  income = HH['population'][i]/max(HH['population'])
  ageP = df1['population'][i]/max(df1['population'])
  hispanic = df2['population'][i]/max(df2['population'])
  percentage_income.append(income)
  percentage_age.append(ageP)
  percentage_hispanic.append(hispanic)

In [0]:
HH['percentage']=percentage_income
df1['percentage']= percentage_age
df2['percentage']= percentage_hispanic

In [0]:
MEAN=[]

In [0]:
for i, row in HH.iterrows():
  m= (HH['percentage'][i]+df1['percentage'][i]+df2['percentage'][i])/3
  MEAN.append(m)
HH['the mean percentage'] = MEAN

In [0]:
print(HH.loc[HH['the mean percentage']==max(HH['the mean percentage'])])

            country   population  percentage  the mean percentage
249  [31000US35620]  1.01121e+06         1.0             0.940082


In [0]:
print(df.loc[df['CBSA']==35620])

      CBSA                                   NAME  POPESTIMATE2017
249  35620  New York-Newark-Jersey City, NY-NJ-PA         20320876


New York-Newark-Jersey City has the highest mean percentage across all three categories with 0.940082